### ISSUE TITLE
incomplete openfold source code

### ISSUE THREAD

**Issue by linghuichen** [jozhang97/MutateEverything#1]
Missing openfold/data and openfold/resources?

**Comment by jozhang97** [comment]
Thanks for your interest in this project.

`openfold/data`: we do not train AlphaFold for structure prediction, instead fine-tune a pre-trained checkpoint.

`openfold/resources`: https://github.com/aqlaboratory/openfold/blob/main/scripts/install_third_party_dependencies.sh might have what you are looking for.

Let me know if you have any more specific questions/issues/problems you are encountering.

**Comment by linghuichen** [comment]
Thanks for your quick reply.

I tried to run the training of AF backbone, but the loss at epoch 19 seems to be much larger than your.

<img width="1386" alt="image" src="https://github.com/jozhang97/MutateEverything/assets/8730660/244a81b8-32b8-4fd9-8821-23e3507fae6b">

The training of ESM backbone looks fine. Is there anything I missed?
(BTW, I dissabled the flash_attn in openfold since I'm not running on A100, this is the only modification I made on the source code).

**Comment by jozhang97** [comment]
Hmm, I am able to reproduce on Quadro RTX 6000.

It looks like the loss is too high. Could you share the entire log? Did you check that you are loading the model correctly?

**Comment by linghuichen** [comment]
I'm runing on Tesla V100. The initial AF checkpoint seems to be loaded correctly.

Here is my config
<img width="1448" alt="image" src="https://github.com/jozhang97/MutateEverything/assets/8730660/c968cef2-3073-4dc9-a00c-3a3b0615ef01">

and the losses from epoch 0:
<img width="1428" alt="image" src="https://github.com/jozhang97/MutateEverything/assets/8730660/b3d90e5e-4a01-4709-a638-9d820246bc7d">

**Comment by jozhang97** [comment]
Thanks for your patience.

Can you try downloading the model we trained and evaluating it? Add the args `--resume /path/to/model --eval`

**Comment by linghuichen** [comment]
> Thanks for your patience.
> 
> Can you try downloading the model we trained and evaluating it? Add the args `--resume /path/to/model --eval`

I ran eval on protherm_multiple using downloaded checkpoing mutate_everything_af.pth, the result looks bad:
<img width="1298" alt="image" src="https://github.com/jozhang97/MutateEverything/assets/8730660/45e344d2-bce4-45ff-9567-df6e32a7f93a">

**Comment by jozhang97** [comment]
Ah, then its probably something different in your openfold installation, can you take a look?

**Comment by linghuichen** [comment]
> Ah, then its probably something different in your openfold installation, can you take a look?

Thanks a lot for your help. This problem may caused by my modification on openfold (I disabled flash attention). But I cannot confirm it because I don't have GPU for flash attention. 
However, I replaced openfold with Uni-Fold and the loss looks fine now.

**Comment by jozhang97** [comment]
That's great! Out of curiosity, was it a lot of work to swap openfold with unifold?

**Comment by linghuichen** [comment]
> That's great! Out of curiosity, was it a lot of work to swap openfold with unifold?

It takes about 1~2 hours to do the swaping. The implementation of openfold and unifold are similar, it's easy to do this.

**Comment by jozhang97** [comment]
Thanks for letting me know! Closing for now, feel free to open another issue.
